controllers/clusterclaims.go

```diff
@@ -0,0 +1,159 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+
+	"github.com/go-logr/logr"
+	odfv1alpha1 "github.com/red-hat-storage/odf-operator/api/v1alpha1"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ClusterClaimKey string
+
+const (
+	OdfVersion            ClusterClaimKey = "odf-version"
+	StorageSystemName     ClusterClaimKey = "odf-storage-system-name"
+	StorageClusterName    ClusterClaimKey = "odf-storage-cluster-name"
+	CephFsid              ClusterClaimKey = "odf-ceph-fsid"
+	RookCephMonSecretName string          = "rook-ceph-mon"
+	FsidKey               string          = "fsid"
+)
+
+type ClusterClaimCreator struct {
+	Context       context.Context
+	Logger        logr.Logger
+	Client        client.Client
+	Values        map[ClusterClaimKey]string
+	StorageSystem *odfv1alpha1.StorageSystem
+}
+
+func (r *StorageSystemReconciler) ensureClusterClaims(instance *odfv1alpha1.StorageSystem, logger logr.Logger) error {
+	ctx := context.TODO()
+	crd := extensionsv1.CustomResourceDefinition{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: "clusterclaims.cluster.open-cluster-management.io"}, &crd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("clusterclaims.cluster.open-cluster-management.io CRD not found. skipping creation of clusterclaims")
```

Comment:
Remove this line, we don't need to spam the logs with this on every reconcile. We should only output when we're actually creating anything.